src/ORM/Table.php

```diff
@@ -1368,6 +1367,37 @@ public function deleteAll($conditions)
     /**
      * {@inheritDoc}
      */
+    public function deleteEach($conditions, $options = [])
+    {
+        $options = $options + [
+                'strict' => true,
+                'limit' => 0
+            ];
+
+        $event = $this->dispatchEvent('Model.beforeDeleteEach', compact('conditions', 'options'));
+        if ($event->isStopped()) {
+            return $event->result();
+        }
+        $conditions = $event->data('conditions');
+
+        $count = 0;
+        do {
+            $entity = $this->find()->where($conditions)->first();
```

Comment:
@ADmad out of memory would be my worry. I should even include options for what fields to load.